src/test/xmeta/test_xmeta_emit.cpp

```diff
@@ -469,38 +503,26 @@ struct ExpectedEnum : ExpectedType
 
 struct ExpectedParam : ExpectedType
 {
-    ElementType type;
     std::string typeRef;
 
-    ExpectedParam(std::string const& expectedName, ElementType const& expectedType)
-        : ExpectedType(expectedName), type(expectedType), typeRef("") {}
     ExpectedParam(std::string const& expectedName, std::string const& expectedType)
-        : ExpectedType(expectedName), type(ElementType::End), typeRef(expectedType) {}
+        : ExpectedType(expectedName), typeRef(expectedType) {}
 
     void VerifyType(Param const& param, ParamSig const& paramSig, std::string_view const& parentNamespace) const override
     {
         REQUIRE(param.Name() == name);
-        if (type == ElementType::End)
-        {
-            VerifyTypeRefName(typeRef, parentNamespace, std::get<coded_index<TypeDefOrRef>>(paramSig.Type().Type()).TypeRef());
-        }
-        else
-        {
-            REQUIRE(std::get<ElementType>(paramSig.Type().Type()) == type);
-        }
+
+        VerifyTypeName(typeRef, parentNamespace, paramSig.Type().Type());
```

Comment:
This interface still lacks the capacity to check things we should already be checking:
- Is a parameter in or out?
- Is it passed by reference?
- Is it an array? (this also applies to return types)

Strongly recommend that instead of adding a pile of arguments to this function, you simply have a ToString for `MethodDef`, `MethodDefSig`, `ParamSig`, `RetTypeSig`, and `TypeSig`, so you only need to compare the expected and actual strings. Then we have something like this ability:
```
ExpectedMethod m{"SomeFunc", "void (Int32[] param1, out ref Object)"};
m.Verify(method_def);
```